Hot fixes for v3

ehannes opened the pull request:
Some hot fixes for v3 :)

- `envfile` parameter is now read correctly
- `PATHS_TO_BACKUP` is correctly read
- Fixes errors in help section
- Removes obsolete `job.template.bash`
- Changes suggestion for how to name passphrase
- Fixes #5

ehannes committed:
Fixes setting envfile

We need to read required parameters before reading the ones that are
optional. As the code was before, envfile was set to `--`.

merikz reviewed (approved):
Det här är bättre, för det är rätt! 👍

ehannes committed:
Fixes reading paths from command line

When we switched to reading envfile from $1 we missed to adjust
`paths_on_cmdline`.

merikz reviewed (changes requested):
Det borde vara en `shift` på rad 87, när vi just använt den. Inte på rad 122

ehannes committed:
Fixes errors in help section

ehannes committed:
Fixes parameter shift after envfile

merikz committed:
Fix creation of *repobase in setup.sh

USER_BACKUP_CONFDIR renamed to CONFDIR.

Missed in commit ffc160: "rename variable USER_BACKUP_CONFDIR"

Fixes #5

ehannes committed:
Removes obsolete job.template.bash

ehannes committed:
Changes suggester naming of passphrase file

I have some backup jobs that has two repositories, for instance one
remote one and one local one, to get a redundant backup. When having
such a setup, you can create a generic env file, for instance
job.env, and then have specific files where you set the repository
path, for intsance job_remote.env and job_local.env. Keeping the host
name in the end of those will give you a much nicer list of jobs in
your .borgbackup. A small detail, yes :)
